### ISSUE TITLE
Hard-coding `width` causes unwelcome consequences on bindings

### ISSUE THREAD

**Issue by basilesimon** [sveltejs/svelte-scroller#13]
The [manual setting of `width`](https://github.com/sveltejs/svelte-scroller/blob/e8544657915f7c4cbf9f97f745827743804bac6f/Scroller.svelte#L88) and its use to set [`svelte-scroller-background-container`'s styles](https://github.com/sveltejs/svelte-scroller/blob/e8544657915f7c4cbf9f97f745827743804bac6f/Scroller.svelte#L98) appears to get in the way of binding to `clientWidth` inside `<Scroller />`.

[Here is a fork of the official example reproducing the behaviour I mention](https://svelte.dev/repl/04252aab939449d2ad58aa363dbce091?version=3.32.1). Notice how the SVG box initially inherits a small width incompatible with its parent's supposed 100%, before this property being naturally altered by, say, a window resize.

**Comment by mhkeller** [comment]
I think I found a solution building off of the previous REPL link. 

**First attempt**

In [this try](https://svelte.dev/repl/1755fc498aae40e9843eae996cc5a441?version=3.35.0), instead of setting the `width:0px` in the `style` variable, I found it to the foreground width. For some reason, that value does not get detected on load and you still get `0`.

**Second attempt**

If you remove the `width` declaration from `style` entirely, it works on load, but without a specific CSS width, the container expands beyond the bounds of its parent when the position switches to `fixed`.

The solution, shown in this [example](https://svelte.dev/repl/2977914e9c4d48f4bbf672fa7a449145?version=3.35.0), we only set the width declaration when the position is `fixed`. I'm not sure why this trigger's svelte's reactivity but it does. We get proper dimension on load because of the regular CSS and then we get it constrained to the parent's bounds when the `fixed` variable changes. 

I wasn't sure if this would fail if you reload the page at a scroll depth inside the scroller but it seems to work. I'll put together a PR.
